polire/idw/idw.py

```diff
@@ -95,13 +105,17 @@ def _predict_grid(self, x1lim, x2lim):
             # New now contains all the points that we're concerned with.
 
         new_grid = np.zeros((len(xx),len(yy)))
-
         for i in range(len(new)):
             # Storing the source data point values into the corresponding grid locations.
             # As from above, we store all the values needed in the new_list. 
             x = new[i][1][0]
             y = new[i][1][1]
-            new_grid[x][y] = new[i][0][2]
+            if new[i][0][2] == 0.0:
+                new_grid[x][y] = 1e-20
```

Comment:
#7 ? Can we not avoid this?

Reply:
new is a list. I can (of course) make it more efficient. I'll update this in the next PR